GitCommands/Git/ExecutableExtensions.cs

```diff
@@ -158,11 +190,23 @@ string ComposeOutput()
         [MustUseReturnValue("Callers should verify that " + nameof(RunBatchCommand) + " returned true")]
         public static bool RunBatchCommand(
             this IExecutable executable,
-            ArgumentString[] batchArguments,
+            ICollection<BatchArgumentItem> batchArguments,
+            Action<BatchProgressEventArgs> action = null,
             byte[] input = null,
             bool createWindow = false)
         {
-            return batchArguments.Aggregate(true, (result, arguments) => executable.RunCommand(arguments, input, createWindow) && result);
+            int total = batchArguments.Sum(item => item.Count);
+            var result = true;
+
+            foreach (var item in batchArguments)
+            {
+                result = executable.RunCommand(item.Argument, input, createWindow) && result;
+
+                // Invoke batch progress callback
+                action?.Invoke(new BatchProgressEventArgs(item.Count, total));
```

Comment:
Should we pass the batch execution result as well?

Reply:
Added. It'll make a better API, though result is not used anywhere ATM.